### ISSUE TITLE
[Vuln] Dapp which rely on lumos could be DoS

### ISSUE THREAD

**Issue by LiRiu** [nervosnetwork/lumos#488]
**Describe the bug**
ckb-indexer rpc get_cells is PREFIX_SEARCH.
if two diff lock_script with same code_hash and same prefix_args like
```
const lock:Script = {
            codeHash:
            "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8",
            hashType: "type",
            args: "0x5f58632a123e7e731db0b59ce50fe16579c7d941",
        }
```
and
```
const lock:Script = {
            codeHash:
            "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8",
            hashType: "type",
            args: "0x5f58632a123e7e731db0b59ce50fe16579c7d941000000000000000000",
        }
```

will be collected unexpect.
so, if dapp use indexer.collector to get cell, and convert them to live cell. it will failed cause of wrong witness for the malform tx which sent by attacker.

**Expected behavior**
help dapp developer to Remove impurities ckb-cell which code_hash is secp256k1 but args is not 20byte.

**Screenshots**
<img width="786" alt="image" src="https://user-images.githubusercontent.com/103174898/221418859-387d9e81-6e3c-4bb6-b61e-7cfdd8f35026.png">

**How to exploit**
victim: 0x5f58632a123e7e731db0b59ce50fe16579c7d941
1. send some CKB to victim_lock_script, but args append some unexpect bytes like 0x5f58632a123e7e731db0b59ce50fe16579c7d94101
2. victim can not send transaction by lumos-indexer-collector

**Comment by homura** [comment]
Maybe we should add the option `script_search_mode` in RPC to support the [exact match](https://github.com/nervosnetwork/ckb-indexer/commit/8cb5a6b127beeb361aa83fc470167b6a2032f15a)

**Comment by LiRiu** [comment]
that's also a feasible solution. 
Seems like we need compare the cost between miners upgrade CKB node with the Dapp developers updating ckb-sdk.

**Comment by LiRiu** [comment]
> Maybe we should add the option `script_search_mode` in RPC to support the [exact match](https://github.com/nervosnetwork/ckb-indexer/commit/8cb5a6b127beeb361aa83fc470167b6a2032f15a)

ohou, seems like the [exact match] mode is [already support](https://github.com/nervosnetwork/ckb-indexer/commit/8cb5a6b127beeb361aa83fc470167b6a2032f15a)

**Comment by LiRiu** [comment]
i try to fix this vuln in PR #490

**Comment by Flouse** [comment]
- link https://github.com/nervosnetwork/ckb/pull/3839
